src/clj_jargon/permissions.clj

```diff
@@ -263,7 +263,23 @@
     write? (.setAccessPermissionWriteInAdminMode dataobj zone fpath user)
     read?  (.setAccessPermissionReadInAdminMode dataobj zone fpath user)))
 
-(defn set-coll-perms
+(defn- set-dataobj-perms-proxied
+  [{^DataObjectAO dataobj :dataObjectAO zone :zone} user fpath read? write? own?]
+  (validate-path-lengths fpath)
+
+  (cond
+    own?   (.setAccessPermissionOwn dataobj zone fpath user)
+    write? (.setAccessPermissionWrite dataobj zone fpath user)
+    read?  (.setAccessPermissionRead dataobj zone fpath user)
+    :else  (.removeAccessPermissionsForUser dataobj zone fpath user)))
```

Comment:
By putting the remove part into the cond, we ensure that the user can't remove their own permissions and then immediately try to set their permissions back to something. Not super likely, but probably better. I kept the admin version the same (remove everything first, then set to whatever's requested), but it could possibly be changed too. Dunno :man_shrugging:

Reply:
Yeah, I'd be inclined to update the admin version to the same, to prevent the slim chance of a crash or other error between the `remove` and subsequent `set`.
Unless there's something `removeAccessPermissionsForUserInAdminMode` will remove that one of those `setAccessPerm` methods will not? 🤷‍♂

Reply:
Yeah, I have no idea why the prior code was done that way. Maybe I'll test it a little more and then change it rather than being overly conservative by not changing it.